pinot-core/src/main/java/org/apache/pinot/core/geospatial/transform/function/BaseBinaryGeoTransformFunction.java

```diff
@@ -0,0 +1,144 @@
+import org.apache.pinot.core.operator.transform.function.TransformFunction;
+import org.apache.pinot.core.plan.DocIdSetPlanNode;
+import org.apache.pinot.segment.local.utils.GeometrySerializer;
+import org.apache.pinot.segment.spi.datasource.DataSource;
+import org.apache.pinot.spi.data.FieldSpec;
+import org.apache.pinot.spi.utils.BytesUtils;
+import org.locationtech.jts.geom.Geometry;
+
+
+/**
+ * Base Binary geo transform functions that can take either one of the arguments as literal.
+ */
+public abstract class BaseBinaryGeoTransformFunction extends BaseTransformFunction {
+  private TransformFunction _firstArgument;
+  private TransformFunction _secondArgument;
+  private Geometry _firstLiteral;
+  private Geometry _secondLiteral;
+  private int[] _intResults;
+  private double[] _doubleResults;
+
+  @Override
+  public void init(List<TransformFunction> arguments, Map<String, DataSource> dataSourceMap) {
+    Preconditions
+        .checkArgument(arguments.size() == 2, "2 arguments are required for transform function: %s", getName());
+    TransformFunction transformFunction = arguments.get(0);
+    Preconditions.checkArgument(transformFunction.getResultMetadata().isSingleValue(),
+        "First argument must be single-valued for transform function: %s", getName());
+    Preconditions.checkArgument(transformFunction.getResultMetadata().getDataType() == FieldSpec.DataType.BYTES
+            || transformFunction instanceof LiteralTransformFunction,
+        "The first argument must be of type BYTES , but was %s",
+            transformFunction.getResultMetadata().getDataType()
+        );
+    if (transformFunction instanceof LiteralTransformFunction) {
+      _firstLiteral = GeometrySerializer.deserialize(
+          BytesUtils.toBytes(((LiteralTransformFunction) transformFunction).getLiteral()));
+    } else {
+      _firstArgument = transformFunction;
+    }
+    transformFunction = arguments.get(1);
+    Preconditions.checkArgument(transformFunction.getResultMetadata().isSingleValue(),
+        "Second argument must be single-valued for transform function: %s", getName());
+    Preconditions.checkArgument(transformFunction.getResultMetadata().getDataType() == FieldSpec.DataType.BYTES
+            || transformFunction instanceof LiteralTransformFunction,
+        "The second argument must be of type BYTES , but was %s",
+            transformFunction.getResultMetadata().getDataType()
+        );
+    if (transformFunction instanceof LiteralTransformFunction) {
+      _secondLiteral = GeometrySerializer.deserialize(
+          BytesUtils.toBytes(((LiteralTransformFunction) transformFunction).getLiteral()));
+    } else {
+      _secondArgument = transformFunction;
+    }
+  }
+
+  @Override
+  public TransformResultMetadata getResultMetadata() {
+    return INT_SV_NO_DICTIONARY_METADATA;
+  }
+
+  @Override
+  public int[] transformToIntValuesSV(ProjectionBlock projectionBlock) {
```

Comment:
(major) Need to first check the result metadata, and only perform the following operations if the metadata is `INT`. Call `super.transformToIntValuesSV(projectionBlock)` if not